src/zcbor_encode.c

```diff
@@ -352,22 +352,34 @@ bool zcbor_tstr_encode(zcbor_state_t *state, const struct zcbor_string *input)
 }
 
 
-bool zcbor_bstr_encode_ptr(zcbor_state_t *state, const char *ptr, size_t len)
+bool zcbor_bstr_encode_ptr(zcbor_state_t *state, const char *str, size_t len)
 {
-	const struct zcbor_string zs = { .value = (const uint8_t *)ptr, .len = len };
+	const struct zcbor_string zs = { .value = (const uint8_t *)str, .len = len };
 
 	return zcbor_bstr_encode(state, &zs);
 }
 
 
-bool zcbor_tstr_encode_ptr(zcbor_state_t *state, const char *ptr, size_t len)
+bool zcbor_tstr_encode_ptr(zcbor_state_t *state, const char *str, size_t len)
 {
-	const struct zcbor_string zs = { .value = (const uint8_t *)ptr, .len = len };
+	const struct zcbor_string zs = { .value = (const uint8_t *)str, .len = len };
 
 	return zcbor_tstr_encode(state, &zs);
 }
 
 
+bool zcbor_bstr_put_term(zcbor_state_t *state, char const *str)
+{
+	return zcbor_bstr_encode_ptr(state, str, strlen(str));
+}
+
+
+bool zcbor_tstr_put_term(zcbor_state_t *state, char const *str)
+{
+	return zcbor_tstr_encode_ptr(state, str, strlen(str));
```

Comment:
Just loose thoughts:
Is it possible to avoid using `strlen()`? It can potentially be unsafe when the string doesn't end with NULL, or the memory isn't accessible. What do you think about handing the responsibility of calculating the string length to the caller by introducing the third argument to specify the length?

Reply:
I agree. I will create a separate PR for that change.